packages/core-app-api/src/lib/AuthConnector/DefaultAuthConnector.ts

```diff
@@ -93,7 +107,11 @@ export class DefaultAuthConnector<AuthSession>
   }
 
   async createSession(options: CreateSessionOptions): Promise<AuthSession> {
-    if (options.instantPopup) {
+    const enableExperimentalRedirectFlow =
+      this.configApi.getOptionalBoolean('enableExperimentalRedirectFlow') ??
+      false;
+
+    if (options.instantPopup && !enableExperimentalRedirectFlow) {
       return this.showPopup(options.scopes);
     }
```

Comment:
```suggestion
    if (options.instantPopup) {
      if (enableExperimentalRedirectFlow) {
        return this.executeRedirect(options.scopes);
      }
      return this.showPopup(options.scopes);
    }
```

There's still the case here where if `instantPopup` is set we want to call `executeRedirect()` straight away, without going through the `authRequester` flow. The `instantPopup` option basically means that the user will be expecting auth to happen immediately as it was initiated through a direct user interaction. When it's not set it might be happening at any point and we'll need to show the modal for the user instead.